File: sdk/python/src/cratebase/admin_service.py
"""Superuser (admin panel) session management."""

from __future__ import annotations

from typing import TYPE_CHECKING

from .types import AdminAuthResponse

if TYPE_CHECKING:
    from .client import Cratebase


class AdminService:
    def __init__(self, client: "Cratebase") -> None:
        self._client = client

    def auth_with_password(self, email: str, password: str) -> AdminAuthResponse:
        result: AdminAuthResponse = self._client.send(
            "/api/admins/auth-with-password",
            method="POST",
            body={"email": email, "password": password},
        )
        self._client.auth_store.save(result["token"], result["admin"])
        return result

    def auth_refresh(self) -> AdminAuthResponse:
        result: AdminAuthResponse = self._client.send("/api/admins/auth-refresh", method="POST")
        self._client.auth_store.save(result["token"], result["admin"])
        return result

File: sdk/go/admin_service.go
package cratebase

// AdminService manages superuser (admin panel) sessions.
type AdminService struct {
	client *Client
}

// AuthWithPassword logs a superuser in with email/password.
func (s *AdminService) AuthWithPassword(email, password string) (*AdminAuthResponse, error) {
	var result AdminAuthResponse
	err := s.client.Send("/api/admins/auth-with-password", SendOptions{
		Method: "POST",
		Body:   map[string]any{"email": email, "password": password},
	}, &result)
	if err != nil {
		return nil, err
	}
	s.client.AuthStore.Save(result.Token, result.Admin)
	return &result, nil
}

// AuthRefresh renews the current superuser session.
func (s *AdminService) AuthRefresh() (*AdminAuthResponse, error) {
	var result AdminAuthResponse
	err := s.client.Send("/api/admins/auth-refresh", SendOptions{Method: "POST"}, &result)
	if err != nil {
		return nil, err
	}
	s.client.AuthStore.Save(result.Token, result.Admin)
	return &result, nil
}
